cpp/rocksdb4pl.cpp

```diff
@@ -64,148 +63,130 @@ typedef enum
   BLOB_FLOAT32,				/* 32-bit IEEE float */
   BLOB_FLOAT64,				/* 64-bit IEEE double */
   BLOB_TERM				/* Arbitrary term */
-} blob_type;
+};
 
-typedef enum
+enum merger_t
 { MERGE_NONE = 0,
   MERGE_LIST,
   MERGE_SET
-} merger_t;
+};
 
-typedef struct dbref
-{ rocksdb::DB	*db;			/* DB handle */
+struct dbref
+{
+  dbref()
+    : db(             nullptr),
+      symbol(         PlAtom(PlAtom::null)),
+      name(           PlAtom(PlAtom::null)),
+      flags(          0),
+      builtin_merger( MERGE_NONE),
+      merger(         PlRecord(PlRecord::null)),
+      type(           { .key   = BLOB_ATOM,
+                        .value = BLOB_ATOM})
+  { }
+
+  rocksdb::DB	*db;			/* DB handle */
+  std::string    pathname;              /* DB's file name (for debugging) */
```

Comment:
pathname should be a PlAtom, because the blob is PL_BLOB_UNIQUE and std::string will break that.